src/process_utility.c

```diff
@@ -695,15 +691,25 @@ process_altertable_change_owner(Hypertable *ht, AlterTableCmd *cmd)
 }
 
 static void
+process_add_constrarains_chunk(Hypertable *ht, Oid chunk_relid, void *arg)
+{
+	Oid		   *hypertable_constraint_oid = (Oid *) arg;
+	Chunk	   *chunk = chunk_get_by_relid(chunk_relid, ht->space->num_dimensions, true);
+
+	chunk_constraint_create_on_chunk(ht, chunk, *hypertable_constraint_oid);
+}
+
+static void
 process_altertable_add_constraint(Hypertable *ht, const char *constraint_name)
 {
 	CatalogSecurityContext sec_ctx;
+	Oid			hypertable_constraint_oid = get_relation_constraint_oid(ht->main_table_relid, constraint_name, false);
 
 	Assert(constraint_name != NULL);
 
 	catalog_become_owner(catalog_get(), &sec_ctx);
 	process_utility_set_expect_chunk_modification(true);
-	process_add_hypertable_constraint(ht, constraint_name);
+	foreach_chunk(ht, process_add_constrarains_chunk, &hypertable_constraint_oid);
```

Comment:
name typo.

Reply:
Btw, I think `process_utility_set_expect_chunk_modification(true)` is now called here outside the loop as well as inside the loop for each individual chunk. I guess it makes most sense to move this stuff as close as possible to the mutating function. This also applies to `catalog_become_owner()`.

This will also become cleaner once we move to a consistent catalog API.

Reply:
Both of the above Fixed